hazelcast/src/test/java/com/hazelcast/internal/cluster/impl/MembershipUpdateTest.java

```diff
@@ -819,6 +827,79 @@ public void testListenerRegistrationWhileANewMemberJoining() {
         assertOpenEventually(entryAddedLatch);
     }
 
+    @Test
+    public void testDistributedObjectsAccurateAfterBatchedJoinHandling() throws InterruptedException {
```

Comment:
can we have an always failing test here? This test is also passing without this PR on my local.

Reply:
> can we have an always failing test here? This test is also passing without this PR on my local.

When running this test on master, it fails for me on every run (a dozen attempts) - I did have occasions where earlier versions would pass during my debugging, but it was still very rare. Does this test pass more than once in a row for you?

Reply:
Looks like the cluster was not forming in some environments (I saw errors related to the port from others who also had it pass), and there was no check for cluster size - I've made some changes in this commit: https://github.com/hazelcast/hazelcast/pull/25114/commits/5869dbffddc94192745b19615eb93a9971ff8ba9

Please try again and let me know 👍